src/analysis/asymptotics/negligible.lean

```diff
@@ -0,0 +1,204 @@
+/-
+Copyright (c) 2021 Devon Tuma. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Devon Tuma
+-/
+import analysis.asymptotics.asymptotics
+import analysis.asymptotics.specific_asymptotics
+import data.polynomial.eval
+
+/-!
+# Negligible Functions
+
+This file defines a predicate `negligible f` on functions `f` from `ℕ` to a normed field `𝕜`.
+
+The main theorem is `negligible_polynomial_mul` that says the product of a polynomial
+  and a negligible function is still a negligible function
+-/
+
+namespace asymptotics
+
+/-- Definition of negligible functions over an arbitrary `normed_field`.
+  Note that the second function always has type `ℕ → ℝ`, which generally gives better lemmas. -/
+def negligible {𝕜 : Type*} [normed_ring 𝕜] (f : ℕ → 𝕜) :=
+∀ (c : ℤ), is_O f (λ n, (n : ℝ) ^ c) filter.at_top
+
+section normed_ring
+
+variables {R : Type*} [normed_ring R]
+variables {f g : ℕ → R}
+
+lemma negligible_of_is_O (hg : negligible g)
+  (h : is_O f g filter.at_top) : negligible f :=
+λ c, h.trans $ hg c
+
+lemma negligible_of_le (hg : negligible g)
+  (h : ∀ n, ∥f n∥ ≤ ∥g n∥) : negligible f :=
+negligible_of_is_O hg (is_O_of_le filter.at_top h)
+
+lemma negligible_of_eventually_le (hg : negligible g)
+  (h : ∀ᶠ n in filter.at_top, ∥f n∥ ≤ ∥g n∥) : negligible f :=
+negligible_of_is_O hg $ is_O_iff.2 ⟨1, by simpa only [one_mul] using h⟩
+
+/-- It suffices to check the negligiblity condition for only sufficiently small exponents `c`.
+  See `negligible_of_is_O_fpow_le` for a version with explicit bounds -/
+lemma negligible_of_eventually_is_O
+  (h : ∀ᶠ (c : ℤ) in filter.at_bot, is_O f (λ n, (n : ℝ) ^ c) filter.at_top) :
+  negligible f :=
+begin
+  obtain ⟨C, hC⟩ := filter.eventually_at_bot.mp h,
```

Comment:
Could you please add `open filter` somewhere near the top of your file, then drop all `filter.`?